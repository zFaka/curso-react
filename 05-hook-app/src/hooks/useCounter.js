import {useState} from 'react';

const useCounter = (initalState = 0) => {


  const [state, setState] = useState(initalState);


  const increase = (factor = 1) => {

    setState(state + factor);

  }


  const decrease = (factor = 1) => {

    setState(state - factor);

  }


  const reset = () => {

    setState(initalState);

  };



  return {

    state, 
    increase, 
    decrease, 
    reset

  };
}

export default useCounter ;
