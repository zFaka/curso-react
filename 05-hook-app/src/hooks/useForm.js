import {useState} from 'react';

const useForm = (initalState = {})=> {


  const [value, setValue] = useState(initalState);

  const handleInputChange = ({target}) => {

    setValue({
      ...value, 
      [target.name]:target.value
    })
  }


  return [value, handleInputChange];
}

export default useForm;
