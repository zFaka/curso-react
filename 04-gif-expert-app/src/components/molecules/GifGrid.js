import React from 'react';
import {useFetchGifs} from '../../hooks/useFetchGifs';
import GifGridItem from '../atoms/GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

  const {data:images, loading} = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeInDown">{category}</h3> 

      {loading && <p className="animate__animated animate__flash">L   o  a ding</p>}

      <div className='card-grid'>
        {
          images.map((img) => (
            <GifGridItem 
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
category: PropTypes.string.isRequired
}
