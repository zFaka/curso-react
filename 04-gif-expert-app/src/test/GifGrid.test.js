import React from 'react';
import {shallow} from "enzyme";
import {GifGrid} from "../components/molecules/GifGrid";
import {useFetchGifs} from '../hooks/useFetchGifs';
jest.mock('../hooks/useFetchGifs')

describe('Testing GifGrid', () => {

  const category = 'Minecraft';



  test('Should render component', () => {

    useFetchGifs.mockReturnValue({
      data:[], 
      loading: true
    });

    const renderOutput = shallow( <GifGrid category={category} />);


    expect(renderOutput).toMatchSnapshot()
  })



  test('Should show items when images are load using useFetchGifs', () => {

    const gifs = [{
      id:'1', 
      title:'whaterevah', 
      url:'https://minecraft.net'
    }];

    useFetchGifs.mockReturnValue({
      data:gifs, 
      loading: false
    });

    const renderOutput = shallow(<GifGrid category={category}/>);


    expect(renderOutput).toMatchSnapshot()
    expect(renderOutput.find('p').exists()).toBe(false);
    expect(renderOutput.find('GifGridItem').length).toBe(gifs.length);
  })



})
