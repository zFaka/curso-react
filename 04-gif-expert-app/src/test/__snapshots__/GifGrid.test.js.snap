// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid Should render component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Minecraft
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    L   o  a ding
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid Should show items when images are load using useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Minecraft
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="whaterevah"
      url="https://minecraft.net"
    />
  </div>
</Fragment>
`;
